src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1241,7 +1241,7 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, CSharpCompi
                 ParameterSymbol receiverParameter = method.Parameters.First();
 
                 // we will have a different receiver if ReplaceTypeOrValueReceiver has unwrapped TypeOrValue
-                if ((object)receiver != receiverArgument)
+                if ((object)receiver != methodGroup.Receiver)
```

Comment:
This is to avoid creating the conversion for the extension method receiver twice, avoiding duplicate "missing compiler-required member" diagnostics.